### ISSUE TITLE
What does Reset Task button really do? - Request for source refresh button

### ISSUE THREAD

**Issue by Horuyuki** [meeb/tubesync#187]
> Reset
If you need to, you can reset and reschedule all tasks using the button below. 

I was wondering if this button will manually scan the sources at the point of clicking? Or does it do something else? I don't get what the reschedule means.

Thank you!

**Comment by meeb** [comment]
Tasks are what TubeSync schedules to perform for actions like downloading videos, indexing channels and getting media metadata like screenshots. All local events in TubeSync. These can be performed just once (like downloading videos) or repeatedly on a timer (like indexing channels and playlists to see if there's any new videos to download).

When a media item or a source is saved any required tasks are created for it, such as if you save a media item and it has no thumbnail then a task will be created to download the thumbnail. Save in this context is a database signal and not something you would proactively do in the interface.

With the current tasks system the tasks on a repeated timer have a tendency stop working for a couple of different reasons.

What the "reset tasks" button does is delete all existing current scheduled tasks in TubeSync and then call the save function on every source and media item. When saved again even with no changes all the required tasks such as indexing channels every X hours are automatically recreated.

In short, the "reset tasks" button is a massive hack to kick the tasks system into behaving again if it stops working as expected and it will be removed once the tasks system isn't quite so broken.

**Comment by Horuyuki** [comment]
I see, do you think a button for manually refreshing a source can be added? That way I don't have to put the whole system on so much load if I just want it to get a few new videos/

**Comment by meeb** [comment]
Possibly, but the general idea is (once it's all working without requiring resetting or other nonsense) you just set and forget it and the downloading is automatic.

**Comment by Horuyuki** [comment]
> Possibly, but the general idea is (once it's all working without requiring resetting or other nonsense) you just set and forget it and the downloading is automatic.

yeah I see where you are coming from, the reason I want a manual refresh is so I can download youtube videos and then download them from the server. Thanks anyways!
